### PR TITLE
feat: Hide code related actions based on model definition size

### PR DESCRIPTION
**Pull request by gt2345**
## Description

Add a new field `modelDefinitionSize` which should be 0 for unmanaged experiments.
Hide code tab and download code action based on `modelDefinitionSize`

## Test Plan

For unmanaged experiments, you should not see code tab or download code action

## Commentary (optional)

WEB-1633

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/ExperimentDetails/ExperimentSingleTrialTabs.tsx [hunk 1]
```diff
@@ -269,17 +269,19 @@ const ExperimentSingleTrialTabs: React.FC<Props> = ({
...
-          <ExperimentCodeViewer
-            experiment={experiment}
-            selectedFilePath={settings.filePath}
-            onSelectFile={handleSelectFile}
-          />
-        ),
-        key: TabType.Code,
-        label: 'Code',
-      });
+      if (experiment.modelDefinitionSize !== 0) {
```
Comment (keita-determined) [inline webui/react/src/pages/ExperimentDetails/ExperimentSingleTrialTabs.tsx:272]: in the ERD meeting, we had 2 options; hide tab or disable tab, and disable tab is the recommended one. Did we choose to hide tab?
We can actually disable tab already if you pass `disabled: true` props btw

File: master/internal/core_experiment.go [hunk 2]
```diff
@@ -391,11 +391,16 @@ func (m *Master) parseCreateExperiment(req *apiv1.CreateExperimentRequest, user
...
-			return nil, config, nil, nil, errors.Wrapf(
-				compressErr, "unable to find compress model definition")
+		if req.ModelDefinition != nil {
+			modelBytes, compressErr = archive.ToTarGz(filesToArchive(req.ModelDefinition))
+			if compressErr != nil {
+				return nil, config, nil, nil, errors.Wrapf(
+					compressErr, "unable to find compress model definition")
+			}
+		} else {
+			modelBytes = []byte{}
```
Comment (NicholasBlaskey) [inline master/internal/core_experiment.go:401]: Do we need this line? Does something later care about the distinction between empty slice vs nil slice?

File: proto/src/determined/experiment/v1/experiment.proto [hunk 3]
```diff
@@ -151,6 +151,8 @@ message Experiment {
   bool unmanaged = 40;
   // Time in seconds which experiment ran or has been running.
   optional int32 duration = 41;
+  // Size of model definition file, for unmanaged experiments this should be 0.
+  optional int32 model_definition_size = 45;
```
Comment (NicholasBlaskey) [inline proto/src/determined/experiment/v1/experiment.proto:155]: Is there a reason for picking 45? I see 42 is taken?

**Review by NicholasBlaskey — COMMENTED** [review]
couple minor comments

backend changes LGTM

File: proto/src/determined/experiment/v1/experiment.proto (hunk 3 above)
Comment (gt2345) [reply proto/src/determined/experiment/v1/experiment.proto:155]: Just merged master, we have 43 and 44 to be external experiment/trial id

File: master/internal/core_experiment.go (hunk 2 above)
Comment (NicholasBlaskey) [reply master/internal/core_experiment.go:401]: Oh ok that makes sense to me

File: webui/react/src/pages/ExperimentDetails/ExperimentSingleTrialTabs.tsx (hunk 1 above)
Comment (gt2345) [reply webui/react/src/pages/ExperimentDetails/ExperimentSingleTrialTabs.tsx]: fixed

**Review by keita-determined — APPROVED** [review]

File: master/internal/core_experiment.go (hunk 2 above)
Comment (keita-determined) [reply master/internal/core_experiment.go:401]: @gt2345 question: why not define `modelBytes := []byte{}` above instead of making if condition?

**Review by NicholasBlaskey — APPROVED** [review]
LGTM

**Pull request merged**